tsdb/shard_test.go

```diff
@@ -73,3 +74,124 @@ func TestShardWriteAndIndex(t *testing.T) {
 		t.Fatalf(err.Error())
 	}
 }
+
+func BenchmarkWritePoints_NewSeries_1K(b *testing.B)   { benchmarkWritePoints(b, 38, 3, 3, 1) }
+func BenchmarkWritePoints_NewSeries_100K(b *testing.B) { benchmarkWritePoints(b, 32, 5, 5, 1) }
+func BenchmarkWritePoints_NewSeries_250K(b *testing.B) { benchmarkWritePoints(b, 80, 5, 5, 1) }
+func BenchmarkWritePoints_NewSeries_500K(b *testing.B) { benchmarkWritePoints(b, 160, 5, 5, 1) }
+func BenchmarkWritePoints_NewSeries_1M(b *testing.B)   { benchmarkWritePoints(b, 320, 5, 5, 1) }
+
+func BenchmarkWritePoints_ExistingSeries_1K(b *testing.B) {
+	benchmarkWritePointsExistingSeries(b, 38, 3, 3, 1)
+}
+func BenchmarkWritePoints_ExistingSeries_100K(b *testing.B) {
+	benchmarkWritePointsExistingSeries(b, 32, 5, 5, 1)
+}
+func BenchmarkWritePoints_ExistingSeries_250K(b *testing.B) {
+	benchmarkWritePointsExistingSeries(b, 80, 5, 5, 1)
+}
+func BenchmarkWritePoints_ExistingSeries_500K(b *testing.B) {
+	benchmarkWritePointsExistingSeries(b, 160, 5, 5, 1)
+}
+func BenchmarkWritePoints_ExistingSeries_1M(b *testing.B) {
+	benchmarkWritePointsExistingSeries(b, 320, 5, 5, 1)
+}
+
+func benchmarkWritePoints(b *testing.B, mCnt, tkCnt, tvCnt, pntCnt int) {
+	// Generate test series (measurements + unique tag sets).
+	series := genTestSeries(mCnt, tkCnt, tvCnt)
+	// Load the in-memory series index.
+	index := NewDatabaseIndex()
+	for _, s := range series {
```

Comment:
actually, this test should be run without building the in-memory index first. I want to see the total time. The actual saving of the metadata to Bolt won't happen if you do this so it skews the result.